delivery/routes.py

```diff
@@ -34,3 +39,29 @@ def put(self, restaurant_id: int):
         restaurant.id = restaurant_id
         updated_restaurant = self.restaurants.update(restaurant)
         return jsonify(self.schema.dump(updated_restaurant)), 200
+
+
+class LoginEndpoint(MethodView):
+    def __init__(self, users, schema):
+        self.users = users
+        self.schema = schema
+
+    def post(self):
+        login = request.json.get('login', None)
+        password = request.json.get('password', None)
+        user_password = self.users.get_user(login).password
+        try:
+            hashed = bcrypt.kdf(password.encode(), salt=login.encode(), desired_key_bytes=len(login), rounds=len(password))
```

Comment:
hm, here they use a much simpler code:
https://github.com/pyca/bcrypt/#password-hashing
what is the motivation for more complex code?

Reply:
I decided to use kdf because I can use login like a salt. This salt will be a constant. May be I don't understand. But when I use random key or key like enviromental variable it can de changed. How can I compare passwords then?

Reply:
1.As far as I understand, salt is encoded into password somehow so that the library can extract it later during comparison.
2. bcrypt.hashpw has 2 arguments: password and salt. So that `login` can be passed as a second argument